src/main/java/nus/climods/logic/commands/AddCommand.java

```diff
@@ -38,19 +42,35 @@ public AddCommand(String moduleCode) {
         toAdd = moduleCode;
     }
 
+    private Module getModule(Model model, String moduleCode) throws CommandException {
+        Optional<Module> optionalModule = model.getModuleList().getListModule(moduleCode);
```

Comment:
Maybe it is better to abstract this out to `ModelManger`? We could add a `ModelManger::getModule` or something